### PR TITLE
Updating to version 3.0.0 of uswds-jekyll

### PR DESCRIPTION
**Pull request by line47**
**Preview:** https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/sw-update-uswds/
- Added h3-h6 to the sample docs page
- Updated available config options for `config`, `header`, and `footer` YML files

#  V3.0.0 updates
## Added
- [Search.gov](https://search.gov/) as the option for search and no longer using [Jekyll pages api search](https://github.com/18F/jekyll_pages_api_search)
- Default 404 page
- Edit this page link option in the footer that can be shown or hidden via a config
- Support for Github Pages and [jekyll-remote-theme](https://github.com/benbalter/jekyll-remote-theme) plugin

## Updated
- USA banner to be it's own component that can be shown or hidden via a config
- USWDS to [version 1.4.5](https://github.com/uswds/uswds/releases/tag/v1.4.5)
- Updated USWDS naming from ["Standards" to "System"](https://designsystem.digital.gov/whats-new/updates/2018/01/17/changing-our-name/)
- Default footer code to align with the [USWDS 'medium' footer](https://components.designsystem.digital.gov/components/detail/footer--default.html)
- NPM task to exclude the USWDS SCSS files

## Removed 
- [Jekyll pages api search](https://github.com/18F/jekyll_pages_api_search) in favor of search.gov 
- Search results layout
- USWDS SCSS assets folder

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -1,6 +1,27 @@
 title: Site title
 
+# GitHub information
```
Comment (jseppi) [inline _config.yml:3]: I like all the explanatory comments in here and in the other config files

File: _data/header.yml [hunk 2]
```diff
@@ -3,12 +3,21 @@
 # Uncomment this statement (by removing the leading '# ') to set the
 # header title differently from the site's title (as defined in
 # _config.yml).
-# title: Header title
+
+title: ""
```
Comment (jseppi) [inline _data/header.yml]: this doesn't quite match with the comment above. Should it be commented out, or an empty string?

File: _data/header.yml [hunk 3]
```diff
@@ -3,12 +3,21 @@
...
+
+# Used to show or hide the official usa banner at the top of the site.
+# Comment out to the following line to hide the usa banner.
```
Comment (jseppi) [inline _data/header.yml]: or set to `false`? Which is preferred?

File: _data/header.yml [hunk 4]
```diff
@@ -3,12 +3,21 @@
...
+usa_banner: true
+
+# this defines the type of header
+# header types can be 'basic' or 'extended'
```
Comment (jseppi) [inline _data/header.yml]: maybe a succinct explanation of `basic` vs `extended` would be nice

**Review by jseppi — CHANGES_REQUESTED** [review]
This looks really good to me.

A couple pretty minor doc things to clean-up in `header.yml` if you wouldn't mind.

File: _data/header.yml (hunk 3 above)
Comment (line47) [reply _data/header.yml]: Either works, I think we chose to comment out as that is the format for most all of the other config options

File: _data/header.yml (hunk 4 above)
Comment (line47) [reply _data/header.yml]: Do you think URLs to the component library would be helpful in the comments? 
https://components.designsystem.digital.gov/components/detail/header--basic.html
https://components.designsystem.digital.gov/components/detail/header--extended.html

Comment (jseppi) [reply _data/header.yml]: oh, that's a good idea!

**Review by jseppi — APPROVED** [review]
🎉

**Comment by line47** [comment]
@wslack - like to get your 👀  on this before merging as well

**Comment by wslack** [comment]
looking now

File: _config.yml (hunk 1 above)
Comment (wslack) [reply _config.yml:3]: agree

File: _config.yml [hunk 5]
```diff
@@ -1,6 +1,27 @@
...
+# This is used for adding an edit this page link to the footer
```
Comment (wslack) [inline _config.yml:4]: are we going to put this in the footer later? If not, not sure we should add it.

**Review by wslack — CHANGES_REQUESTED** [review]
Love the documentation.

I think we need more work on the new footer. It seems a lot more unpredictable than the previous one at various widths.

**Comment by wslack** [comment]
(to be clear - I'm referring to the footer on the docs page)

File: _data/footer.yml [hunk 6]
```diff
@@ -1,26 +1,65 @@
...
 heading: Name of agency
 
-# An array of agency logos to show, side by side, in the lower left
-# of the footer on large screens.
+# Configuration for agency logo(s) (shown side by side).
+# If the logo is external add external: true
 logos:
-  - src: /assets/img/us_seal.png
-    alt: Your agency name
+  - src: https://18f.gsa.gov/assets/img/logos/18f-logo.svg
```
Comment (wslack) [inline _data/footer.yml]: why not use an internal 18F logo and an external GSA logo to show both ways?

File: _config.yml (hunk 5 above)
Comment (line47) [reply _config.yml:4]: We can show the "edit this page" link by default, is that what you are suggesting?

File: _data/footer.yml (hunk 6 above)
Comment (line47) [reply _data/footer.yml]: We can add both logos by default - that's an easy update.

**Comment by line47** [comment]
@wslack can you explain a bit more about how the footer is unpredictable?
We moved from a custom footer to using the USWDS medium (default) footer, maybe there are a few things that could be addressed upstream?

https://components.designsystem.digital.gov/components/detail/footer--default.html

File: _data/footer.yml (hunk 6 above)
Comment (wslack) [reply _data/footer.yml]: Let's do that - to show both ways to link an image.

**Comment by wslack** [comment]
@line47 the behavior of the footer at various page widths is wonky. It jumps up and down and gets out of sorts.

**Comment by wslack** [comment]
I can't figure out what I was referring to. I don't like how the footer looks on the documentation page (it appears too similar to the menu) but I don't understand "unprediable.' Maybe was related to using a large screen at the office. I think you can discard.

File: _config.yml (hunk 5 above)
Comment (wslack) [reply _config.yml:4]: Yes! If we ask for the info in the config file, we should implement it or put the code in with a comment.

**Comment by line47** [comment]
Updated to add a the GitHub link to the footer, will need to add functionality for multiple logos in the footer to the uswds-jekyll. At the moment the images are right against each other and need some visual separation. I've added an issue for this https://github.com/18F/uswds-jekyll/issues/127 

@wslack I think this is ready for another pass 👍

**Comment by wslack** [comment]
That image for the GSA logo doesn't work for me. @line47

File: _data/footer.yml [hunk 7]
```diff
@@ -1,26 +1,67 @@
...
 heading: Name of agency
 
-# An array of agency logos to show, side by side, in the lower left
-# of the footer on large screens.
+# Configuration for agency logo(s) (shown side by side).
+# If the logo is external add external: true
 logos:
-  - src: /assets/img/us_seal.png
-    alt: Your agency name
+  - src: /assets/img/gsa-logo-dark.jpg
```
Comment (wslack) [inline _data/footer.yml:34]: Looks like the file needs to be dropped into https://github.com/18F/federalist-uswds-template/tree/master/assets/img

**Review by wslack — CHANGES_REQUESTED** [review]

File: _data/footer.yml (hunk 7 above)
Comment (line47) [reply _data/footer.yml:34]: Thanks for that - I've added the second logo (GSA). We should be good to go!

**Review by wslack — APPROVED** [review]

**Pull request merged**